src/js/ui/multiPortSelectorView.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+    // TODO: Add support for virtual ports.
+    // TODO: Add support for matching based on portSpecs rather than just id.
+    // TODO: Disentangle the desired port spec from the selected port spec.
+    fluid.defaults("youme.multiPortSelectorView", {
+        gradeNames: ["fluid.viewComponent"],
+
+        selectBoxLabel: "MIDI Port:",
+
+        desiredPortSpecs: [],
+
+        model: {
+            ports: [],
+            selectedPortIds: [],
+            portSpecs: []
+        },
+
+        components: {
+            system: {
+                type: "youme.system"
+            },
+            portConnector: {
+                type: "youme.multiPortConnector",
+                options: {
+                    model: {
+                        portSpecs: "{multiPortSelectorView}.model.portSpecs"
+                    }
+                }
+            },
+
+            selectBox: {
+                type: "youme.multiSelectBox",
+                container: "{that}.container",
+                options: {
+                    model: {
+                        label: "{youme.multiPortSelectorView}.options.selectBoxLabel",
+                        selectedItemIds: "{youme.multiPortSelectorView}.model.selectedPortIds",
+                        optionItems: "{youme.multiPortSelectorView}.model.ports"
+                    }
+                }
+            }
+        },
+
+        modelListeners: {
+            ports: {
+                funcName: "youme.multiPortSelectorView.findDesiredPorts",
+                args: ["{that}"]
+            },
+            selectedPortIds: {
+                funcName: "youme.multiPortSelectorView.updatePortSpecs",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    youme.multiPortSelectorView.findDesiredPorts = function (that) {
```

Comment:
This one can also be converted from the effectful modelListener to modelRelay style

Reply:
Done.